### ISSUE TITLE
Page Cache Compatibility Mode 

### ISSUE THREAD

**Issue by joshin42** [szepeviktor/fix-w3tc#39]
Howdy,

When upgrading sites that from 0.9.4.1 to today's pull, I encountered an error when compatibility mode was enabled. Our environment works fine with it off, so I have a work around.

I can replicate this on multiple sites and servers. The error looks like this:

[Tue Aug 30 18:15:28 2016] [crit] [client 24.56.43.116] configuration error:  couldn't perform authentication. AuthType not set!: /wp-content/cache/page_enhanced/www.clientsite.co**m//c**ategory/fai**th//_in**dex.html_gzip, referer: http://www.clientsite.com/

The pair of //'s in the path seem to be the culprit. 

Sorry that I suck at coding or I'd be more useful and contribute fixes. I'm happy to test the latest code on some busy sites I manage & write up interesting things I find.

-J

**Comment by nigrosimone** [comment]
@amiga-500, maybe the problem was introduced with this commit https://github.com/szepeviktor/fix-w3tc/commit/954b739a3bffe3e6d61b044029f4f4da0f269036 (https://github.com/szepeviktor/w3-total-cache-fixed/pull/24)

I think that the problem is:

```
Require all granted
```

`Require all granted` is only in 2.4, but `Allow all XXX` stops working in 2.4.  `authz_host_module` was introduced in [Apache 2.1](https://httpd.apache.org/docs/2.2/en/mod/mod_authz_host.html).  The generated code:

```
<Files ~ "\.(html|html_gzip|xml|xml_gzip)$">
  <IfModule mod_authz_host.c>
    Require all granted
  </IfModule>
  <IfModule !mod_authz_host.c>
    Allow from all
  </IfModule>
</Files>
```

is not safe on Apache < 2.4 with module `mod_authz_host` loaded.

I found some info on [stackoverflow](http://stackoverflow.com/questions/21265191/apache-authtype-not-set-500-error).

**Step to reproduce**
1. Enable compatibility mode in page cache settings
2. Enable mod_authz_host on Apache < 2.4

**Possible fix**

```
<Files ~ "\.(html|html_gzip|xml|xml_gzip)$">
  <IfModule mod_authz_host.c>
    <IfVersion < 2.4>
      Allow from all
    </IfVersion>
    <IfVersion >= 2.4>
      Require all granted
    </IfVersion>
  </IfModule>
  <IfModule !mod_authz_host.c>
    Allow from all
  </IfModule>
</Files>
```

but this require `mod_version` module

---

@joshin42, thanks for your testing!

**Comment by charlesLF** [comment]
I think @nigrosimone is right.

Here's what I propose:

`<IfModule mod_version.c>
	<IfVersion < 2.4>
		Order Deny,Allow
		Deny from All
	</IfVersion>
	<IfVersion >= 2.4>
		Require all denied
	</IfVersion>
</IfModule>
<IfModule !mod_version.c>
	<IfModule !mod_authz_core.c>
		Order Deny,Allow
		Deny from All
	</IfModule>
	<IfModule mod_authz_core.c>
		Require all denied
	</IfModule>
</IfModule>`

**Comment by charlesLF** [comment]
I think @nigrosimone is right.

Here's what I would suggest we use:

```
<IfModule mod_version.c>
	<IfVersion < 2.4>
		Order Deny,Allow
		Deny from All
	</IfVersion>
	<IfVersion >= 2.4>
		Require all denied
	</IfVersion>
</IfModule>
<IfModule !mod_version.c>
	<IfModule !mod_authz_core.c>
		Order Deny,Allow
		Deny from All
	</IfModule>
	<IfModule mod_authz_core.c>
		Require all denied
	</IfModule>
</IfModule>
```

**Comment by joshin42** [comment]
Agreed with the diagnosis, thanks! :) 

Yes, we're still on 2.2, Sites with compatibility mode are (even our) edge cases & we can disable it on them without problem

Thankfully, we'll finally be moving to 2.4 in a month or two.

**Comment by nigrosimone** [comment]
@joshin42 i have applied @charlesLF suggestion. Can you test the master?

**Comment by joshin42** [comment]
Thanks @nigrosimone, it seems to be working fine now!

-J

**Comment by nigrosimone** [comment]
@joshin42 Thanks!

**Comment by amiga-500** [comment]
sorry my mistake. and the belated reply.  And thx for the fix .  I noticed when i had written  e4efc3f4f4f3f66842d6208c65b79f1eacad961e i apparently didnt notice _mod_authz_host_ was also available in 2.2...yikes!  When i researched this at the time it seems i was looking incorrectly under 2.0 modules (mod_authz_host doesnt exist for it) mistakenly thinking i was researching all things pre-2.4. whoops! 

thx again
Kimberly

**Comment by nigrosimone** [comment]
No problem Kimberly! ; )
